src/query/api/v1/handler/placement/add.go

```diff
@@ -115,7 +131,17 @@ func (h *AddHandler) Add(
 		return nil, err
 	}
 
-	service, algo, err := ServiceWithAlgo(h.client, httpReq.Header)
+	serviceOpts := NewServiceOptionsFromHeaders(serviceName, httpReq.Header)
+	if serviceName == M3AggServiceName {
+		if req.MaxAggregationWindowSizeNanos == 0 || req.WarmupDurationNanos == 0 {
+			return nil, errAggWindowAndWarmupMustBeSet
+		}
+		serviceOpts.M3Agg = &M3AggServiceOptions{
+			MaxAggregationWindowSize: time.Duration(req.MaxAggregationWindowSizeNanos),
+			WarmupDuration:           time.Duration(req.WarmupDurationNanos),
```

Comment:
Could we not have a default warmup duration and max aggregation window size if they don't specify any?  I wouldn't have a good idea of what to choose for either of these without reading something first, and most users could probably change this later/shouldn't need to worry too much about these values perhaps?

Reply:
We can probably have a sane default for warmupDuration, but maxAggregationWindowSize is very difficult to pick a sane default value for because if it’s too short they could lose data and if it’s too long their deploy takes forever